test/ParallelIOTest.cpp

```
@@ -1865,4 +1865,161 @@ TEST_CASE("unavailable_backend", "[core][parallel]")
     }
 #endif
 }
+
+void joined_dim(std::string const &ext)
+{
+    using type = float;
+    using patchType = uint64_t;
+    constexpr size_t patches_per_rank = 5;
+    constexpr size_t length_of_patch = 10;
+
+    int size{-1};
+    int rank{-1};
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    {
+        Series s(
+            "../samples/joinedDimParallel." + ext,
+            Access::CREATE,
+            MPI_COMM_WORLD);
+        std::vector<UniquePtrWithLambda<type>> writeFrom(patches_per_rank);
+
+        auto it = s.writeIterations()[100];
+
+        Dataset numParticlesDS(determineDatatype<patchType>(), {0});
+        numParticlesDS.joinedDimension = 0;
+        auto numParticles =
+            it.particles["e"]
+                .particlePatches["numParticles"][RecordComponent::SCALAR];
+        auto numParticlesOffset =
+            it.particles["e"]
+                .particlePatches["numParticlesOffset"][RecordComponent::SCALAR];
+        numParticles.resetDataset(numParticlesDS);
+        numParticlesOffset.resetDataset(numParticlesDS);
+
+        auto patchOffset = it.particles["e"].particlePatches["offset"]["x"];
+        auto patchExtent = it.particles["e"].particlePatches["extent"]["x"];
+        Dataset particlePatchesDS(determineDatatype<float>(), {0});
+        particlePatchesDS.joinedDimension = 0;
```

Is a sanity check needed here? do we want to throw error if someone set an invalid value of joinedDimension?

---

We will probably switch to the same API as ADIOS2, i.e. have some magic number `JOINED_DIMENSION=std::numeric_limits<>::max` and then use something like `Dataset ds(<type>, {JOINED_DIMENSION, 128 , 128})`. We should add a sanity check there.

---

The constructor of `Dataset` now verifies this.